src/gui/generalsettings.cpp

```diff
@@ -30,21 +30,107 @@
 
 #include "ignorelisteditor.h"
 #include "common/utility.h"
+#include "logger.h"
 
 #include "config.h"
 
 #include "legalnotice.h"
 
+#include <QFileDialog>
+#include <QMessageBox>
 #include <QNetworkProxy>
 #include <QDir>
 #include <QScopedValueRollback>
 
+#include <private/qzipwriter_p.h>
+
 #define QTLEGACY (QT_VERSION < QT_VERSION_CHECK(5,9,0))
 
 #if !(QTLEGACY)
 #include <QOperatingSystemVersion>
 #endif
 
+namespace {
+struct ZipEntry {
+    QString localFilename;
+    QString zipFilename;
+};
+
+ZipEntry fileInfoToZipEntry(const QFileInfo &info)
+{
```

Comment:
If I would have done this you likely would have said: This should be handled in a dedicated class - right? 😜

Reply:
Actually not. :-)

I was pondering moving it in a non anonymous namespace in a separate file but decided not to because it's used only there. In any case that doesn't ask for a class IMO.

Reply:
So I think I'll keep it like this. I don't see any good reason to make it more public yet or to move away from a function.